lightgbm/src/main/scala/com/microsoft/ml/spark/lightgbm/PartitionProcessor.scala

```diff
@@ -86,4 +153,13 @@ object PartitionProcessor {
       validDatasetOpt.foreach(_.close())
     }
   }
+
+  def resetState(useSingleDatasetMode: Boolean, isEnabledWorker: Boolean, log: Logger,
+                 sharedState: SharedState): Unit = {
+    if (isEnabledWorker && useSingleDatasetMode) {
+      log.info(s"LightGBM enabled task exiting and resetting state")
+      // Clear the singleton pool and reset the dataset singleton state
+      sharedState.resetSingletonDatasetState()
```

Comment:
done, removed the code to clear the pool in comment above

Reply:
it looks like you are trying to manage a single shared state across multiple calls to fit. Instead, you can make the shared singleton within the fit method and then just throw it away instead of resetting it and managing it. The shared singleton is almost like a hybrid of a class and an object in this sense

Reply:
done!

Reply:
I dont understand this looks like its still in the PR

Reply:
sorry misunderstood the comment, I removed resetSingletonDatasetState method now